### ISSUE TITLE
Mailbox user can't delete messages

### ISSUE THREAD

**Issue by guestisp** [freescout-helpdesk/freescout#205]
A mailbox user is unable to delete messages on it's mailbox. It's able to only delete assigned messages to him.

I think that any unassigned messages should be deletable by any mailbox users, only deletion of messages assigned to other users should be blocked.

**Comment by freescout-helpdesk** [comment]
Yes.

**Comment by guestisp** [comment]
Anyone working on this ?

**Comment by freescout-helpdesk** [comment]
No.

**Comment by guestisp** [comment]
I've looked at this issue, but freescout permissions for conversations are still unclear to me.

@freescout-helpdesk are you able to work on this ? we are able to delete (unassigned) conversation only using the administrator account and this is very annoying.

What is strange is that we get a success notification but the message is still shown

**Comment by freescout-helpdesk** [comment]
> A mailbox user is unable to delete messages on it's mailbox. It's able to only delete assigned messages to him.

Users are able to delete any conversations when "Users are allowed to delete notes/conversations" is set in Settings.

**Comment by guestisp** [comment]
Ok, so if user doesn't have that permission the delete button should be hidden and currently, with delete button shown, freescout show a success warning even if deletion doesn't succeed

**Comment by freescout-helpdesk** [comment]
Delete button is displayed always to avoid extra DB request for permissions check every time conversation is opened.

**Comment by guestisp** [comment]
This is misleading as you are showing a non-working button to users (very bad for usability and UX) and clicking it will return a success message even without permossioms

**Comment by freescout-helpdesk** [comment]
For now priority here is given to performance. When mailbox user permissions caching will be implemented, button will be hidden.

**Comment by guestisp** [comment]
**ONE** query (just **one**, not 1000) over a unique index (the `name` field on `options` table has a unique index, as it should be) doesn't hurt any performance but improves UX a lot and fix a bug because this is a bug. You can't return a success message if the operation wasn't ok.

**Comment by guestisp** [comment]
The issue is related when using the bulk delete because this code:
```
foreach ($conversations as $conversation) {
   if (!$user->can('delete', $conversation)) {
      continue;
   }
```

will always succeed, the deletion is skipped but no warning are shown to the user.

**Comment by freescout-helpdesk** [comment]
https://github.com/freescout-helpdesk/freescout/issues/229

**Comment by guestisp** [comment]
Fixed #229
